src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/CancelAzureSqlDatabaseActivity.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.Database.Cmdlet
+{
+    [Cmdlet(VerbsLifecycle.Stop, "AzureRmSqlDatabaseActivity", SupportsShouldProcess = true,
```

Comment:
@jiawang003 this might be done elsewhere, so please correct me if I'm wrong, but where do you check if the user set the `-WhatIf` parameter in the cmdlet call? Usually this is done by wrapping the execution code in your cmdlet with the following:

```cs
if (ShouldProcess(target, actionMessage)
{
    // cmdlet execution
}
```

Reply:
fixed by overriding ExecuteCmdlet and handles ShouldProcess